src/FunctionApps/python/IntakePipeline/__init__.py

```diff
@@ -1,46 +1,114 @@
+    AzureFhirserverCredentialManager,
+)
+from IntakePipeline.conversion import (
+    convert_batch_messages_to_list,
+    convert_message_to_fhir,
 )
 
 from config import get_required_config
 
 from phdi_transforms.geo import get_smartystreets_client
 
 
-def run_pipeline():
+def run_pipeline(
+    bundle: dict,
+    datatype: str,
+    cred_manager: AzureFhirserverCredentialManager,
+):
     salt = get_required_config("HASH_SALT")
     geocoder = get_smartystreets_client(
         get_required_config("SMARTYSTREETS_AUTH_ID"),
         get_required_config("SMARTYSTREETS_AUTH_TOKEN"),
     )
-    fhirserver_cred_manager = get_fhirserver_cred_manager(
-        get_required_config("FHIR_URL")
+
+    transform_bundle(geocoder, bundle)
+    add_patient_identifier(salt, bundle)
+    store_bundle(
+        get_required_config("INTAKE_CONTAINER_URL"),
+        get_required_config("VALID_OUTPUT_CONTAINER_PATH"),
+        bundle,
+        datatype,
+    )
+    upload_bundle_to_fhir_server(cred_manager, bundle)
+
+
+def convert_to_fhir(
+    blob: func.InputStream,
+    fhir_url: str,
+    container_url: str,
+    invalid_output_path: str,
+    cred_manager: AzureFhirserverCredentialManager,
+) -> Tuple[str, dict]:
+    if blob.name[-3:].lower() not in ("hl7", "xml"):
+        raise Exception(f"invalid file extension for {blob.name}")
+
+    filetype = blob.name.split("/")[-2].lower()
+
+    if filetype == "elr":
+        message_type = "oru_r01"
+        message_format = "hl7v2"
+        bundle_type = "ELR"
+    elif filetype == "vxu":
+        message_type = "vxu_v04"
+        message_format = "hl7v2"
+        bundle_type = "VXU"
+    elif filetype == "eICR":
+        message_type = "ccda"
+        message_format = "ccda"
+        bundle_type = "ECR"
+    else:
+        raise Exception(f"Found an unidentified message_format: {filetype}")
+
+    # VA sends \\u000b & \\u001c in real data, ignore for now
+    messages = convert_batch_messages_to_list(
+        blob.read().decode("utf-8", errors="ignore")
     )
 
-    container_url = get_required_config("INTAKE_CONTAINER_URL")
-    container_prefix = get_required_config("INTAKE_CONTAINER_PREFIX")
-    output_path = get_required_config("OUTPUT_CONTAINER_PATH")
+    for i, message in enumerate(messages):
+        token = cred_manager.get_access_token()
+        response = convert_message_to_fhir(
+            message=message,
+            message_format=message_format,
+            message_type=message_type,
+            access_token=token.token,
+            fhir_url=fhir_url,
+        )
 
-    for datatype, bundle in read_fhir_bundles(container_url, container_prefix):
-        transform_bundle(geocoder, bundle)
-        add_patient_identifier(salt, bundle)
-        store_bundle(container_url, output_path, bundle, datatype)
-        upload_bundle_to_fhir_server(fhirserver_cred_manager, bundle)
+        if response.status_code == 200:
+            return bundle_type, response.json()
```

Comment:
Are these values reversed from what's expected in line 110? It's unpacking the values expecting the bundle to be first, but we're returning it as the second value here. Also, I think we should change line 110 to use bundle_type to keep the variable names consistent.

Reply:
Yeah we definitely can't have a `return` here with a singular value. Based on line 107 where we iterate over a sequence of bundles, it could make sense to create a list of tuples of `(str, dict)`, i.e. `valid_processed_bundles: List[Tuple[str, dict]]. Then when we process an individual bundle, we just access the ith tuple.